hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/impl/TokenInfoPrecompile.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.services.store.contracts.precompile.impl;
+
+import com.hedera.services.config.NetworkInfo;
+import com.hedera.services.state.submerkle.ExpirableTxnRecord;
+import com.hedera.services.store.contracts.WorldLedgers;
+import com.hedera.services.store.contracts.precompile.SyntheticTxnFactory;
+import com.hedera.services.store.contracts.precompile.codec.DecodingFacade;
+import com.hedera.services.store.contracts.precompile.codec.EncodingFacade;
+import com.hedera.services.store.contracts.precompile.utils.PrecompilePricingUtils;
+import com.hedera.services.store.contracts.precompile.utils.TokenInfoRetrievalUtils;
+import com.hederahashgraph.api.proto.java.TokenID;
+import com.hederahashgraph.api.proto.java.TransactionBody.Builder;
+import java.util.function.UnaryOperator;
+import org.apache.tuweni.bytes.Bytes;
+
+public class TokenInfoPrecompile extends AbstractTokenInfoPrecompile {
+
+    public TokenInfoPrecompile(
+            final TokenID tokenId,
+            final SyntheticTxnFactory syntheticTxnFactory,
+            final WorldLedgers ledgers,
+            final EncodingFacade encoder,
+            final DecodingFacade decoder,
+            final PrecompilePricingUtils pricingUtils,
+            final NetworkInfo networkInfo) {
+        super(tokenId, syntheticTxnFactory, ledgers, encoder, decoder, pricingUtils, networkInfo);
+    }
+
+    @Override
+    public Builder body(Bytes input, UnaryOperator<byte[]> aliasResolver) {
+        final var tokenInfoWrapper = decoder.decodeGetTokenInfo(input);
+        tokenId = tokenInfoWrapper.tokenID();
+        return super.body(input, aliasResolver);
+    }
+
+    @Override
+    public Bytes getSuccessResultFor(final ExpirableTxnRecord.Builder childRecord) {
+        return encoder.encodeGetTokenInfo(
```

Comment:
Is there a reason we are going through `WorldLedgers` instead of the much more effient `com.hedera.services.context.primitives.StateView#infoForToken`?  If so it should be documented somewhere.  Probably here.

Reply:
The logic now uses `StateView#infoForToken` and `StateView#infoForNft` methods.